q2_demux/_transformer.py

```diff
@@ -0,0 +1,39 @@
+import tempfile
+import shutil
+import os.path
+
+import skbio
+from q2_types.per_sample_sequences import FastqGzFormat
+
+from .plugin_setup import plugin
+from ._demux import BarcodeSequenceIterator
+from ._format import EMPMultiplexedDirFmt, EMPMultiplexedSingleEndDirFmt
+
+
+@plugin.register_transformer
+def _1(dirfmt: EMPMultiplexedDirFmt) -> BarcodeSequenceIterator:
+    barcode_generator = skbio.io.read(
+        str(dirfmt.barcodes.view(FastqGzFormat).path), format='fastq',
+        constructor=skbio.DNA, phred_offset=33)
+    sequence_generator = skbio.io.read(
+        str(dirfmt.sequences.view(FastqGzFormat).path), format='fastq',
+        constructor=skbio.DNA, phred_offset=33)
+    result = BarcodeSequenceIterator(barcode_generator, sequence_generator)
+    # ensure that dirfmt stays in scope as long as result does so these
+    # generators will work.
+    result.__dirfmt = dirfmt
+    return result
+
+
+@plugin.register_transformer
+def _2(dirfmt: EMPMultiplexedSingleEndDirFmt) -> EMPMultiplexedDirFmt:
+    result = tempfile.mkdtemp()
```

Comment:
Use this so that the framework can manage where the temporary directory lives (once we have a qiime temp dir):

```python
result = EMPMultiplexedDirFmt().path

sequences_fp = result / 'sequences.fastq.gz'
barcodes_fp = result / 'barcodes.fastq.gz'
shutil.copyfile(str(dirfmt.sequences.view(FastqGzFormat)), sequences_fp)
shutil.copyfile(str(dirfmt.barcodes.view(FastqGzFormat)), barcodes_fp)

return result
```